src/main/java/com/connexta/search/index/IndexService.java

```diff
@@ -6,33 +6,16 @@
  */
 package com.connexta.search.index;
 
-import com.connexta.search.index.exceptions.IndexException;
 import java.io.InputStream;
-import java.util.Set;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
 
-/** Interfaces with the index provider. */
 public interface IndexService {
 
-  /**
-   * Indexes the resource.
-   *
-   * @param datasetId the id of the resource
-   * @param mediaType the type of the resource
-   * @param inputStream the resource
-   * @throws IndexException if there was an error indexing the resource
-   */
   void index(
```

Comment:
One thing I don't like about the relationship between this class and SearchManager is their need to both enforce argument constraints in their APIs. I could see these easily getting out of sync.

Reply:
So far, we've been following the structure where each Controller has one corresponding Service. This follows that model. I think the Services and Manger will evolve over time to not be delegation.

Reply:
@peterhuffer @emmberk I looked at more Spring Boot projects the last few months. I saw that controllers take the simple types (strings, numbers, streams, and lists) and use them to create new domain objects like Index. The controller then passes domain objects to the service. Something to think about for the future.

Reply:
Yeah eventually we will need to do this otherwise we will have very large method arguments depending on how much we index.